word_language_model/model.py

```diff
@@ -55,3 +57,165 @@ def init_hidden(self, bsz):
                     weight.new_zeros(self.nlayers, bsz, self.nhid))
         else:
             return weight.new_zeros(self.nlayers, bsz, self.nhid)
+
+class TransformerSeq2Seq(nn.Module):
+    r"""A transformer model applied for sequence-to-sequence transform.
+        User is able to modified the attributes as needed.
+    Args:
+        src_vocab: the number of vocabularies in the source sequence (required).
+        tgt_vocab: the number of vocabularies in the target sequence (required).
+        d_model: the dimension of the encoder/decoder embedding models (default=512).
+        nhead: the number of heads in the multiheadattention models (default=8).
+        num_encoder_layers: the number of sub-encoder-layers in the encoder (default=6).
+        num_decoder_layers: the number of sub-decoder-layers in the decoder (default=6).
+        dim_feedforward: the dimension of the feedforward network model (default=2048).
+        dropout: the dropout value (default=0.1).
+    Examples::
+        >>> transformer_model = nn.Transformer(src_vocab, tgt_vocab)
+        >>> transformer_model = nn.Transformer(src_vocab, tgt_vocab, nhead=16, num_encoder_layers=12)
+    """
+
+    def __init__(self, src_vocab, tgt_vocab, d_model=512, nhead=8, num_encoder_layers=6,
```

Comment:
It could be cleaner to merge this into RNNModel, rename that to SeqModel and then have a flag that you can use to branch into each, as long as the forward methods can accept the same inputs and give the same outputs.